### PR TITLE
[e2e ingress-gce] Add test for pre-shared certificate

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**:
Add an e2e test for the pre-shared certificate feature (`ingress.gcp.kubernetes.io/pre-shared-cert`). Also made some changes to ingress_util.go mostly for certificate generation and polling on ingress.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #NONE 

**Special notes for your reviewer**:
/assign @rramkumar1 @nicksardo 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/network/ingress.go [hunk 1]
```diff
@@ -153,10 +155,49 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
...
+					if err := gceCloud.DeleteSslCertificate(preSharedCertName); err != nil {
+						return false, nil
+					}
+					return true, nil
+				})
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to delete ssl certificate %q: %v", preSharedCertName, err))
+			}()
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create ssl certificate %q: %v", preSharedCertName, err))
+
+			By("Creating an ingress referenceing the pre-shared certificate")
```
Comment (nicksardo) [inline test/e2e/network/ingress.go]: nit: small typo: referenceing

File: test/e2e/network/ingress.go [hunk 2]
```diff
@@ -153,10 +155,49 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
...
+				Certificate: string(cert),
+				PrivateKey:  string(key),
+				Description: "pre-shared cert for ingress testing",
+			})
+			defer func() {
+				By(fmt.Sprintf("Deleting ssl certificate %q on GCE", preSharedCertName))
+				// We would not be able to delete the cert until ingress controller
+				// cleans up the target proxy that references it.
+				err := wait.Poll(framework.LoadBalancerPollInterval, framework.LoadBalancerCleanupTimeout, func() (bool, error) {
+					if err := gceCloud.DeleteSslCertificate(preSharedCertName); err != nil {
```
Comment (nicksardo) [inline test/e2e/network/ingress.go]: Check for NotFound?

File: test/e2e/network/ingress.go [hunk 3]
```diff
@@ -153,10 +155,49 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
...
+			gceCloud, err := framework.GetGCECloud()
+			Expect(err).NotTo(HaveOccurred())
+			_, err = gceCloud.CreateSslCertificate(&compute.SslCertificate{
+				Name:        preSharedCertName,
+				Certificate: string(cert),
+				PrivateKey:  string(key),
+				Description: "pre-shared cert for ingress testing",
+			})
+			defer func() {
+				By(fmt.Sprintf("Deleting ssl certificate %q on GCE", preSharedCertName))
```
Comment (MrHohn) [inline test/e2e/network/ingress.go]: In fact, we need to delete the ingress before deleting the cert. Will fix.

File: test/e2e/network/ingress.go [hunk 4]
```diff
@@ -153,10 +155,49 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
...
+			Expect(err).NotTo(HaveOccurred())
+			gceCloud, err := framework.GetGCECloud()
+			Expect(err).NotTo(HaveOccurred())
+			_, err = gceCloud.CreateSslCertificate(&compute.SslCertificate{
+				Name:        preSharedCertName,
+				Certificate: string(cert),
+				PrivateKey:  string(key),
+				Description: "pre-shared cert for ingress testing",
+			})
+			defer func() {
```
Comment (nicksardo) [inline test/e2e/network/ingress.go]: I'd move the defer to before the CreateSslCertificate as long as you check for NotFound in the delete. It's a bit difficult to read having the create asserts so far away from the create.

File: test/e2e/network/ingress.go (hunk 1 above)
Comment (MrHohn) [reply test/e2e/network/ingress.go]: Done.

File: test/e2e/network/ingress.go (hunk 2 above)
Comment (MrHohn) [reply test/e2e/network/ingress.go]: Sure, done.

File: test/e2e/network/ingress.go (hunk 3 above)
Comment (MrHohn) [reply test/e2e/network/ingress.go]: Done.

File: test/e2e/network/ingress.go (hunk 4 above)
Comment (MrHohn) [reply test/e2e/network/ingress.go]: Make sense, done.

File: test/e2e/framework/ingress_utils.go [hunk 5]
```diff
@@ -1122,28 +1152,31 @@ func (j *IngressTestJig) WaitForIngress(waitForNodePort bool) {
...
+	var cert []byte
+	if len(j.Ingress.Spec.TLS) > 0 {
+		knownHosts = j.Ingress.Spec.TLS[0].Hosts
+		cert = j.GetRootCA(j.Ingress.Spec.TLS[0].SecretName)
+	}
+	j.PollIngressWithCert(waitForNodePort, knownHosts, cert)
+}
+
+// WaitForIngress waits till the ingress acquires an IP, then waits for its
+// hosts/urls to respond to a protocol check (either http or https). If
```
[the remark this answers is not in the document]
Comment (MrHohn) [reply test/e2e/framework/ingress_utils.go]: Thanks, done.

**Comment by MrHohn** [comment]
@nicksardo @bowei Rebased, need re-lgtm, thanks!

**Pull request merged**